sled-agent/src/instance.rs

```diff
@@ -584,7 +627,22 @@ impl InstanceRunner {
                                 tx.send(Ok(self.current_state()))
                                     .map_err(|_| Error::FailedSendClientClosed)
                             },
-                            PutState{ state, tx } => {
+                            PutState { state, tx } => {
+                                // If we're going to stop the instance, start
+                                // the timeout after which we will forcefully
+                                // terminate the VMM.
+                                if let VmmStateRequested::Stopped = state {
+                                    // Only start the stop timeout if there
+                                    // isn't one already, so that additional
+                                    // requests to stop coming in don't reset
+                                    // the clock.
```

Comment:
It does seem that we'd call `put_state` again in this case though -- is that desirable?

(a valid answer may be: "Sure!" - I'm just confirming)

Reply:
🤷‍♀️ we would call it twice before this change, so I didn't mess with it. I don't see any obvious reason why it would be *bad* to do it again, and maybe it's useful for e.g. a higher-level component retrying?

Reply:
FWIW my general philosophy here has been to have sled agent pass all state change requests right through to Propolis and let Propolis decide how to deal with any duplicates. (In this case it should just decide the second stop request can be ignored because it's already put one on its queue.)

Reply:
Yeah, that makes sense to me. I thought not resetting the timeout was worthwhile, though: you can imagine a scenario where an instance fails to stop, and a frustrated user or retrying client keeps sending more stop requests, each of which resets the grace period and stops the stuck VMM from being killed.

Reply:
Oh, I definitely agree--once the sled agent finds out about the control plane's intent to stop the instance, it should arm the timer and not reset it if the same intent is expressed again. Mostly I just wanted to provide a bit of color about calling Propolis twice; the Propolis state machine is designed to handle that kind of thing precisely so that sled agent doesn't have to worry about it.

Reply:
Yup, thanks for that! I figured we were on the same page about the timer but felt like it was worth clarifying further.